ilastik/app.py

```diff
@@ -256,6 +260,27 @@ def _update_hbp_mode(parsed_args):
         ilastik_config.set("ilastik", "hbp", "true")
 
 
+def _update_tiktorch(parsed_args, root_path: str):
+    """enable tiktorch local workflow"""
+    tiktorch_executable: Optional[Path] = None
+    if parsed_args.tiktorch_executable:
+        tiktorch_executable = Path(parsed_args.tiktorch_executable)
+    else:
+        # Maybe tiktorch is bundled
+        root_path = Path(root_path)
+        tiktorch_script = "run_tiktorch.sh" if platform.system() != "Windows" else "run_tiktorch.bat"
+        bundled_tiktorch_executable = root_path / "tiktorch" / tiktorch_script
+
+        if bundled_tiktorch_executable.exists():
+            tiktorch_executable = bundled_tiktorch_executable
+
+    if tiktorch_executable and tiktorch_executable.exists():
+        tiktorch_msg = "Using tiktorch executable: %s" % tiktorch_executable
+        print(tiktorch_msg)
+        logger.info(tiktorch_msg)
+        ilastik_config.set("ilastik", "tiktorch_executable", str(tiktorch_executable))
```

Comment:
maybe it's not a good idea to safe this in config at all. It should be determined at runtime - either via the argument or via detecting it in the distro (or from ilastikrc) but I wouldn't write there. Or do you think that the benefits outweigh the potential issues?

Reply:
I think we could introduce a singleton object for this kind of options, only reason it's in config right now is because it looked similar to hbp and debug flags.